staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -237,14 +250,120 @@ func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attr
 		return nil
 	}
 	if len(errs) > 1 {
-		for i := 1; i < len(errs); i++ {
-			// TODO: merge status errors; until then, just return the first one.
-			utilruntime.HandleError(errs[i])
-		}
+		return aggregateWebhookErrors(errs)
 	}
 	return errs[0]
 }
 
+func aggregateWebhookErrors(errs []error) error {
+	if len(errs) == 1 {
+		return errs[0]
+	}
+
+	errorsByReason := groupErrorsByReason(errs)
+
+	// When there are multiple groups of errors, we can only return one group.
+	// We use priorities to decide the group to return. The items are ordered from
+	// highest to lowest priority. We prioritize "actionable" errors, i.e. errors
+	// that can be resolved by some user action, like changing the resource.
+	for _, reason := range statusReasonsByPriorityDescending {
+		if group, exists := errorsByReason[reason]; exists {
+			mergedErr := mergeErrorsOfSameType(group)
+			if statusErr, ok := mergedErr.(*apierrors.StatusError); ok { //nolint:errorlint
+				return statusErr
+			}
+			return apierrors.NewInternalError(mergedErr)
+		}
+	}
+	return errs[0] // fallback
+}
+
+func groupErrorsByReason(errs []error) map[metav1.StatusReason][]error {
```

Comment:
rather than fully building the `map[reason]list` for all errors/reasons, and then throwing away all but one of the list entries, can we iterate once to find the `minimum index, most prioritized reason` entry, then iterate a second time from that index, accumulating mergeable errors?